#!/usr/bin/env python
PACKAGE = "shootPlanning"

import roslib;
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("alwaysLobshot", bool_t, 0, "Always call the Lobshot", False)
gen.add("alwaysStraightshot", bool_t, 0, "Always call the Straightshot", False)
gen.add("shotStrengthScaling", double_t, 0.0, "Scale factor to reduce shot power", 1.0, 0, 1)
gen.add("shotAngleScaling", double_t, 0.0, "Scale factor to reduce straightshot angle", 1.0, 0, 1)
gen.add("lobAngleScaling", double_t, 0.0, "Scale factor to reduce lobshot angle", 1.0, 0, 1)
gen.add("passScaling", double_t, 0.0, "Scale factor to reduce pass power", 1.0, 0, 1)
gen.add("minLobDistance", double_t, 0.0, "min distance from goal to be able to lob", 0.0, 0, 20)
gen.add("maxLobDistance", double_t, 0.0, "max distance from goal to be able to lob", 0.0, 0, 20)
    
exit(gen.generate(PACKAGE, "ShootPlanningNode", "ShootPlanningNode"))

  
